src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -184,6 +184,53 @@ class RAY_EXPORT GcsClient : public std::enable_shared_from_this<GcsClient> {
   std::function<void()> resubscribe_func_;
 };
 
+// This client is only supposed to be used from Cython / Python
+class RAY_EXPORT GcsSyncClient {
+ public:
+  explicit GcsSyncClient(const GcsClientOptions &options);
+  Status Connect();
```

Comment:
The cpp GCS client has Disconnect API which shuts down the underlying rpc client. Maybe we should call the Disconnect upon Python client `__del__`?

Reply:
So I looked into this and it is not necessary / there is no explicit way to terminate or disconnect channels. We have shutdown logic in our C++ code to drain the event loops etc. since that code is asynchronous, but for the synchronous code no disconnect logic is needed :)

The shutdown logic seems to be happening in the channel destructor: https://github.com/grpc/grpc/blob/ffafac3ce8ea53cbceffb53063070be813edc31f/src/cpp/client/channel_cc.cc#L56

Reply:
that sounds good. do you mind adding comments regarding this to the code ? (that disconnect & channel clean up is done by grpc)